saltboot-formula/_states/saltboot.py

```diff
@@ -1896,6 +1896,20 @@ def _check_terminal_kernel_parameters(parameters):
         msg = 'OK'
     return ok, msg
 
+def _get_salt_pid():
+    pid_file = '/var/run/venv-salt-minion.pid'
+    res = __salt__['cmd.run_all']("ls /usr/bin/venv-salt-minion", python_shell=True)
+    if res['retcode'] > 0:
+        pid_file = '/var/run/salt-minion.pid'
+    res = __salt__['cmd.run_all']("cat {0}".format(pid_file), python_shell=True)
+    pid = res['stdout']
+    return pid
```

Comment:
Why do you do this through `cmd` module and not `file`?
We have good functions for it [file.file_exists](https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.file.html#salt.modules.file.file_exists) and [file.read](https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.file.html#salt.modules.file.read)

Reply:
And for writing [file.write](https://docs.saltproject.io/en/latest/ref/modules/all/salt.modules.file.html#salt.modules.file.write)